statistics/median.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Returns the arithmetic median of the numbers in the given collection
+ *
+ * Example:
+ *
+ * ```ts
+ * import { assertEquals } from "../testing/asserts.ts"
+ * import { median } from "./median.ts"
+ *
+ * const numbers = [ 4, 2, 7 ]
+ * assertEquals(median(numbers), 4)
+ * ```
+ */
+export function median(collection: Array<number>): number | undefined {
+  if (collection.length === 0) {
+    return undefined;
+  }
+
+  const middle = Math.floor(collection.length / 2);
+  const values = Array.from(collection);
+  values.sort((a, b) => a - b);
+
+  if (values.length % 2 === 0) {
+    return (values[middle] + values[middle - 1]) / 2;
```

Comment:
This was a little confusing to me since you're essentially getting the two values in reverse order that they appear in the set... I know it doesn't matter mathematically but it would have helped me at least with readability to reverse them like:

```ts
return (values[middle-1] + values[middle]) / 2
```

Or maybe:
```ts
return mean(values.slice(middle - 1, middle + 1))
```